samples/HttpRequestFeatures/HttpRequestFeaturesExample/Controllers/HomeController.cs

```diff
@@ -23,12 +22,19 @@ public IActionResult Index()
         {
             HttpRequestFeatures httpRequestFeatures = GetHttpRequestFeaturesFromRequest(Request);
 
-            ViewData["UserAgent"] = JsonConvert.SerializeObject(httpRequestFeatures, Formatting.Indented);
+            ViewData["UserAgent"] = JsonConvert.SerializeObject(httpRequestFeatures, Formatting.Indented);
+            
+            // Generate an ID to associate with the request.
+            string eventId = Guid.NewGuid().ToString();
+            ViewData["EventId"] = eventId;
 
-            Tuple<string, string, string> personalizationobj = callPersonalizationService(httpRequestFeatures);
-            ViewData["Personalizer Rank Request"] = personalizationobj.Item1;
-            ViewData["Personalizer Rank Response"] = personalizationobj.Item2;
-            ViewData["Personalizer rewardActionId"] = personalizationobj.Item3;
+            Tuple<string, string, string> personalizerRank = callPersonalizerRank(httpRequestFeatures, eventId);
+            ViewData["Personalizer Rank Request"] = personalizerRank.Item1;
+            ViewData["Personalizer Rank Response"] = personalizerRank.Item2;
+            ViewData["Personalizer rewardActionId"] = personalizerRank.Item3;
+
+            string personalizerReward = callPersonalizerReward(eventId);
+            ViewData["Personalizer Reward Request"] = personalizerReward;

```

Comment:
This is teaching the customer a bad behavior. Please make sure there is an action that occurs between receiving a rank and sending a reward. Otherwise, new users will automatically send a reward without knowing why.